c10/macros/Macros.h

```diff
@@ -109,6 +109,12 @@ namespace at { namespace cuda { using namespace c10::hip; }}
 #define C10_HOST_DEVICE __host__ __device__
 #define C10_DEVICE __device__
 #define C10_HOST __host__
+#define C10_MAX_THREADS_PER_BLOCK(val) (((val) <= 1024) ? (val) : (512))
```

Comment:
@iotamudelta Hey, can you check if these numbers are right for ROCm? We probably need different numbers in that case.

Reply:
I'd like a comment here on these macros, exactly analogous to what you wrote in the commit message here.

Reply:
@ezyang technically, ROCm supports up to 2048 work group size. Assuming the caller logic indeed never calls with more than 1024 threads, this should be fine for us.